### PR TITLE
Str to int

### PR DESCRIPTION
**Pull request by gvarao**
- Added fields in rules to plot field data
- Modified field types from str to integer

### CODE REVIEW

File: juniper_official/Linecard/lo-statistics.rule [hunk 1]
```diff
@@ -16,6 +16,18 @@ healthbot {
                     frequency 60s;
                 }
             }
+            field target{
+                sensor center-chip-lookup-out {
+                    path target;
+                }
+                type string;
+            }
+            field cchip-lookup-out-errors{
```
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/lo-statistics.rule]: Spacing..

File: juniper_official/Linecard/scheduler-info.rule [hunk 2]
```diff
@@ -15,6 +15,36 @@ healthbot {
                     frequency 60s;
                 }
             }
+            field target{
+                sensor fpc-cpu-scheduler {
+                    path target;
+                }
+                type string;
+            }
+            field cpu-1-second{
```
Comment (ConstrainedRandomVar) [inline juniper_official/Linecard/scheduler-info.rule]: Spacing here and below..

File: juniper_official/System/pfe-traffic-statistics.rule [hunk 3]
```diff
@@ -70,26 +90,26 @@ healthbot {
                 frequency 60s;
                 term are-host-path-drops-increasing {
                     when {
-                        min-rate-of-increase "$host-path-hardware-drops" { ## Warning: 'min-rate-of-increase' is deprecated
+                        min-rate-of-increase "$host-path-hardware-drop-count" { ## Warning: 'min-rate-of-increase' is deprecated
```
Comment (ConstrainedRandomVar) [inline juniper_official/System/pfe-traffic-statistics.rule]: Can we move away from min-rate-of-increase to incrasing-at-least-by?

**Comment by gvarao** [comment]
> Looks good.
> The first batch of rule files have fields with descriptions, but these are skipped in the later rule files. Would be good to track this through a subsequent cleanup effort to ensure that all our factory default rules have field definitions.. Needn't be done now.

Venkat- I am not sure about field functionality as these rules are from JTAC. I will open PR to add field descriptions by next HB release.

File: juniper_official/Linecard/scheduler-info.rule (hunk 2 above)
Comment (gvarao) [reply juniper_official/Linecard/scheduler-info.rule]: Addressed

File: juniper_official/Linecard/lo-statistics.rule (hunk 1 above)
Comment (gvarao) [reply juniper_official/Linecard/lo-statistics.rule]: Addressed

File: juniper_official/Linecard/fi-statistics.rule [hunk 4]
```diff
@@ -58,7 +94,7 @@ healthbot {
                 frequency 60s; 
                 term are-center-chip-fabric-in-errors-present {
                     when {
-                        greater-than "$cchip-fi-packets-with-error" 0;
+                        greater-than "$cchip-fi-error-packets" 0;
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Linecard/fi-statistics.rule]: Thanks

File: juniper_official/Linecard/check-pfe-discards.rule [hunk 5]
```diff
@@ -29,27 +29,97 @@ healthbot {
...
                 }
                 type string;
                 description "FPC slot number";
             }
+            field info-cell-discard {
+                sensor pfe-sensor {
+                    path info-cell-discard;
+                }
+                type integer;
+                description "Infor cell discard count";
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Linecard/check-pfe-discards.rule]: Addresses

File: juniper_official/Kernel/krtasync.rule [hunk 6]
```diff
@@ -6,31 +6,43 @@ healthbot {
...
                 }
+                type integer;
+                description "KRT tree insert count";
             }
             field krtq-async-count {
                 reference {
                     path "/topic[topic-name=krt.statistics]/rule[rule-name=check-krt-state]/krtq-async-count";
                 }
+                type integer;
+                description "KRT async count";
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Kernel/krtasync.rule]: Addressed

File: juniper_official/Kernel/krtasync.rule [hunk 7]
```diff
@@ -193,15 +234,26 @@ healthbot {
...
+                }
+                type string;
+                description "KRT queue state";
+            }
             field krtq-async-count {
                 sensor krtstate {
                     path Krt-Async-Count;
                 }
+                type integer;
+                description "KRT async count";
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Kernel/krtasync.rule]: Addressed

File: juniper_official/Chassis/system-virtual-memory.rule [hunk 8]
```diff
@@ -15,16 +14,30 @@ healthbot {
                     frequency 60s;
                 }
             }
+            field interrupt-rate {
+                sensor system-virtual-memory {
+                    path interrupt-rate;
+                }
+                type integer;
+            }
+            field system-kmem-map-free{
```
[the remark this answers is not in the document]
Comment (gvarao) [reply juniper_official/Chassis/system-virtual-memory.rule]: addressed

File: juniper_official/System/pfe-traffic-statistics.rule (hunk 3 above)
Comment (gvarao) [reply juniper_official/System/pfe-traffic-statistics.rule]: We will address through PR as change is required in many rules so we can do changes in all rule in one go.

**Pull request merged**